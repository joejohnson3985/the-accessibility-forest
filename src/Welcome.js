import React, { Component } from 'react';
import './Welcome.scss';

class Welcome extends Component {
  constructor(props) {
    super(props);
    this.state = {
      name: ''
    }
  }

  handleName = (e) => {
    e.preventDefault();
    this.props.nameForest(this.state.name);
  }

  handleChange = (e) => {
    this.setState({
      name: e.target.value
    })

  }

  render() {
    return (
      <div>
        <form className='name-forest-form' onSubmit={this.handleName}>
          <input className='name-forest-input' value={this.state.name} onChange={this.handleChange} type='text' placeholder='Name Your Forest'/>
          <input className ='submit-btn' type='submit' />
        </form>
      </div>
    )
  }
}

export default Welcome;
