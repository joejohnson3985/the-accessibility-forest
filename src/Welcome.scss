@import './_variables.scss';

.name-forest-input {
  margin-right: 10px;
  padding: 8px;
  font-size: 20px;
  border-radius: 4px;
  border: none;
  color: $primary-color;
}

.submit-btn {
  @include primary-button;

  &:hover {
    @include hover-btn;
  }

  &:active {
    @include active-btn
  }
}