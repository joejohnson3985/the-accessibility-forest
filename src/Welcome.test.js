import React from 'react';
import Welcome from './Welcome';
import { shallow } from 'enzyme';

const mockFunc = jest.fn();
const e = {target: {value: 'test'}};

describe('Welcome', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
        <Welcome nameForest={mockFunc}/>
      )
  });

  it('Should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it.skip('Should update state of name on change', () => {
    expect(wrapper.state('name')).toEqual('')
    wrapper.find('.name-forest').simulate('change', {target: {value: 'test'}})
    expect(wrapper.state(name)).toEqual('test')
  })

  it('Should call submit name on click', () => {
    wrapper.find('.name-forest-form').simulate('submit', {preventDefault: () => {} })
    expect(mockFunc).toBeCalled();
  })

})