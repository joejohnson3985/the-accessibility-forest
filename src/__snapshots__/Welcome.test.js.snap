// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome Should match the snapshot 1`] = `
<div>
  <form
    className="name-forest-form"
    onSubmit={[Function]}
  >
    <input
      className="name-forest-input"
      onChange={[Function]}
      placeholder="Name Your Forest"
      type="text"
      value=""
    />
    <input
      className="submit-btn"
      type="submit"
    />
  </form>
</div>
`;
